tidy3d/plugins/microwave/custom_path_integrals.py

```diff
@@ -0,0 +1,163 @@
+"""Helper classes for performing custom path integrals with fields on the Yee grid"""
+
+from __future__ import annotations
+
+import pydantic.v1 as pd
+import numpy as np
+import xarray as xr
+
+from typing import Literal, Union
+
+from ...constants import MICROMETER
+from ...components.data.monitor_data import FieldData, FieldTimeData, ModeSolverData
+from ...components.data.data_array import (
+    ScalarFieldDataArray,
+    ScalarFieldTimeDataArray,
+    ScalarModeFieldDataArray,
+)
+from ...components.data.data_array import FreqDataArray, TimeDataArray, FreqModeDataArray
+from ...components.base import cached_property
+from ...components.types import Axis, ArrayFloat2D
+from ...exceptions import DataError, SetupError
+
+from .path_integrals import AbstractAxesRH
+
+MonitorDataTypes = Union[FieldData, FieldTimeData, ModeSolverData]
+EMScalarFieldType = Union[ScalarFieldDataArray, ScalarFieldTimeDataArray, ScalarModeFieldDataArray]
+IntegralResultTypes = Union[FreqDataArray, FreqModeDataArray, TimeDataArray]
+FieldParameter = Literal["E", "H"]
+
+
+class CustomPathIntegral2D(AbstractAxesRH):
+    """Class for defining a custom path integral defined as a curve on an axis-aligned plane."""
```

Comment:
Will be helpful to provide a formula here explaining how path integral is performed; I think it's based on trapezoidal rule, and thus 2nd order convergence?

Reply:
Here is added explanation rendered in the docs:

![image](https://github.com/flexcompute/tidy3d/assets/155486263/e474ec85-70c0-4d03-9c6e-7b1a33e0571f)

Reply:
nice